clients/src/main/java/org/apache/kafka/clients/admin/AlterConsumerGroupOffsetsResult.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin;
+
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.annotation.InterfaceStability;
+import org.apache.kafka.common.internals.KafkaFutureImpl;
+
+/**
+ * The result of the {@link AdminClient#alterConsumerGroupOffsets(String, Map)} call.
+ *
+ * The API of this class is evolving, see {@link AdminClient} for details.
+ */
+@InterfaceStability.Evolving
+public class AlterConsumerGroupOffsetsResult {
+
+    private final KafkaFutureImpl<Map<TopicPartition, KafkaFutureImpl<Void>>> future;
+
+    public AlterConsumerGroupOffsetsResult(KafkaFutureImpl<Map<TopicPartition, KafkaFutureImpl<Void>>> future) {
+        this.future = future;
+    }
+
+    public KafkaFutureImpl<Map<TopicPartition, KafkaFutureImpl<Void>>> values() {
```

Comment:
I don't think we need the nested futures here. The new api just works with a single group, so seems like the type should just be `KafkaFuture<Map<TopicPartition, Void>>`. Also, note that we don't want to return `KafkaFutureImpl` directly.

Reply:
Yes. For consistency, it's actually best to have `KafkaFuture<Map<TopicPartition, Errors>>`. So it's the same as `deleteConsumerGroupOffsets()`.

Reply:
Huhmm... Actually I think that's a mistake in `deleteConsumerGroupOffsets`. We don't want to expose `Errors` directly. I will submit a separate PR

Reply:
https://issues.apache.org/jira/browse/KAFKA-8992

Reply:
Let's just use the same two APIs from `deleteConsumerGroupOffsets`:
```java
    public KafkaFuture<Void> partitionResult(final TopicPartition partition);
    public KafkaFuture<Void> all();
```

Reply:
This is a good catch, thanks @hachikuji , we can address 8992 within the 2.4 deadline.